### DISCUSSION TITLE
BTNS Subassets

### DISCUSSION THREAD

**Discussion by jdogresorg** [jdogresorg/Broadcast-Token-Naming-System#20] [Ideas]
A few users have expressed interest in getting subasset functionality implemented into BTNS. It has also been mentioned that the BTNS Subasset system should follow the XCP style architecture with a protected ASSET namespace (ASSET.SUBASSET).

I too would like to see a subasset system implemented into BTNS, however, I do not feel that copying what is already working within XCP is the right path forward for BTNS, for a number of reasons
- Period `.` is already allowed in BTNS TICK names
- BTNS Tokens with `.` have already been issued
- BTNS has been created to play with new token functionality and concepts, not just copy what already exists.

I see definite value in having the ability to create a token which is a related to another token in a `parent->child` style relationship.

## Counterparty / XCP Subasset Format
The current counterparty `SUBASSET` format limits the `ASSET` namespace to 12 characters, allows the `SUBASSET` to be up to 250 characters, and requires the `ASSET` name be included in the `SUBASSET` name. 

### FORMAT
ASSET 
->ASSET.SUBASSET

### Examples
PIZZA
-> PIZZA.Dominos
-> PIZZA.Dominos.2023.PROMOS.August.BOGO
-> PIZZA.Dominos.2023.PROMOS.July_Pizza_Discount
-> PIZZA.Hut
-> PIZZA.PapaJohns
-> PIZZA.PapaJohns-Coupon-August-BOGO
-> PIZZA.PaPaJohn-Ownership-Shares

## BTNS Subasset Format
The general idea I have for the BTNS `SUBASSET` format is to allow users to issue a `SUBASSET` which is directly related a parent `TICK`. 

### FORMATS
TICK
-> SUBASSET

### Examples
PIZZA
-> Dominos
-> Dominos.2023.PROMOS.August.BOGO
-> Dominos.2023.PROMOS.July_Pizza_Discount
-> PizzaHut
-> PapaJohns
-> PapaJohns-Coupon-August-BOGO
-> PaPaJohn-Ownership-Shares

This proposed format limits subassets to 1 level deep, but allows users to establish their own "subasset' naming system, just like in Counterparty.

## Notes
Ideally would like to allow for infinite recursion of `parent->child` tokens, so that PIZZA could issue PIZZA->Dominos subasset and then Dominos would be able to issue all their own subassets PIZZA->Dominoes->Subassets without having to rely on PIZZA owner to issue. 

However,  this would require passing the full TICK for every level deep, which could very quickly get out of control cost wise having to broadcast a HUGE string just to specify a token name... Need to chat with Javier a bit more about this idea before moving forward with subassets.

If we allow infinite `parent->child` subasset functionality, it will be __VERY__ important that the explorer clearly indicate the FULL path to the subasset tokens, to help users differentiate between subasset tokens of similar names.

**Comment by ghost** [comment]
BTNS SubAsset  naming system 

📁 Amazon eGift Card
└📁 amazon.com gift card
⋯└ 📁 Reload
⋯⋯└ 🖼️$20
⋯⋯└ 🖼️$50
⋯⋯└ 🖼️$100
⋯└ 📁 Amazon.co.jp Gift card
⋯⋯└ 🖼️$20
⋯⋯└🖼️ $50
⋯⋯└ 🖼️$100
└📁 Vincent van Gogh
⋯└ 📁 NFT
⋯⋯└ 🖼️ The Starry Night

**Reply by jdogresorg** [reply]
This would be ideal... tho have some issues to address first... see "Notes" section in original post :)
